syntax = "proto3";

package network.events;

//protoc --python_out=. network_event_extended.proto

message NetworkEvent {
    string event_id = 1;
    int64 timestamp = 2;
    string source_ip = 3;
    string target_ip = 4;
    int32 packet_size = 5;
    int32 dest_port = 6;
    int32 src_port = 7;
    string agent_id = 8;
    float anomaly_score = 9;
    double latitude = 10;
    double longitude = 11;

    // Campos existentes para eventos enriquecidos cuando alguien trate de usar TOR.
    string event_type = 12;
    float risk_score = 13;
    string description = 14;

    // NUEVO: Identificador del SO para selección de reglas
    string so_identifier = 15;  // "linux_ufw", "linux_iptables", "windows_firewall", "darwin_pf"

    // NUEVO: Información adicional del nodo (opcional, solo en primer evento)
    string node_hostname = 16;        // Hostname del nodo
    string os_version = 17;           // "Ubuntu 22.04", "Windows 11", etc.
    string firewall_status = 18;      // "active", "inactive", "unknown"
    string agent_version = 19;        // Versión del agente
    bool is_initial_handshake = 20;   // true solo en el primer evento del nodo
}